app/mutations/supply/complete.rb

```diff
@@ -7,8 +7,13 @@ class Supply::Complete < Mutations::Command
   def execute
     supply.complete = true
 
-    add_error :complete, :failed unless supply.save
+
+    if supply.save
+      Task::Comments::Base.run(task: supply, message: 'completed', user: user)
```

Comment:
This works, but I'm always confused when I see a Something::Base class used directly. I consider them abstract, i.e. I wouldn't use them directly but only inherit and use the child class.